### ISSUE TITLE
request failed, status: 500 Internal Server Error

### ISSUE THREAD

**Issue by N1k-3l** [grafana/grafana-zabbix#1356]
**Describe the bug**
When try to get metrics from group of hosts (more than 50) --> _Error 'request failed, status: 500 Internal Server Error'_
Can it be from plugin error, because with old (3version) plugin it worked normal?

**Log**

```
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Publication successful" logger=live user=16 channel=grafana/dashboard/uid/dtx8-Ednk
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Querying for data source via SQL store" logger=datasources id=5 orgId=1
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Initializing new data source instance" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Zabbix request" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource method=apiinfo.version
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Got Zabbix version" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource version=54.000
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Zabbix request" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource method=apiinfo.version
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Zabbix request" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource method=hostgroup.get
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Successfully authenticated" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource url=http://ZABBBIX_XXX/api_jsonrpc.php user=XXX
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Zabbix request" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource method=hostgroup.get
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Writing result to cache" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource method=hostgroup.get
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Zabbix request" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource method=host.get
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Writing result to cache" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource method=host.get
t=2021-12-04T16:09:53+0300 lvl=dbug msg="Zabbix request" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource method=item.get
t=2021-12-04T16:09:59+0300 lvl=eror msg="Zabbix API request error" logger=plugins.backend pluginId=alexanderzobnin-zabbix-datasource error="request failed, status: 500 Internal Server Error"
t=2021-12-04T16:09:59+0300 lvl=eror msg="Request Completed" logger=context userId=16 orgId=1 uname=XXXXX method=POST path=/api/datasources/5/resources/zabbix-api status=500 remote_addr=XXX.XXX.XXX.XXX time_ms=5375 size=95 referer="https://grafana.XXX.ru/d/dtx8-Ednk/network-interfaces-utilization?orgId=1&editPanel=4"
```

**Screenshots**

`request failed, status: 500 Internal Server Error
Object
status:500
statusText:"Internal Server Error"
data:Object
error:"Internal Server Error"
message:"request failed, status: 500 Internal Server Error"
config:Object
url:"api/datasources/5/resources/zabbix-api"
method:"POST"
headers:Object
hideFromInspector:false
data:Object
retry:0
message:"request failed, status: 500 Internal Server Error"`
![image](https://user-images.githubusercontent.com/74911055/144710454-29b0a548-6405-4953-aba9-66f599d92f93.png)

![image](https://user-images.githubusercontent.com/74911055/144713099-b140b8cb-5328-4beb-838c-b7700316fc5b.png)

**Software versions**

| Grafana | Zabbix | Grafana-Zabbix Plugin |
| ------- | ------ | --------------------- |
| 8.2.5   | 5.4.8  | 4.2.4                 |

**Comment by mbuyukkarakas** [comment]
Hello.

**Comment by alexanderzobnin** [comment]
It looks like Zabbix API cannot handle request. Try to tweak PHP settings and increase amount of memory assigned to PHP. I appreciate if you can make the same request in grafana with 3.x plugin version to make sure it's something that works in a different way in the new plugin versions with backend.

**Comment by middilebeast** [comment]
Hello @alexanderzobnin 
why it was working before with the same php settings as am using now ?
does the amount of data affects the API Call ? 
Regards

**Comment by middilebeast** [comment]
By the way our php memory settings are 10 G and we still has the same issue

**Comment by N1k-3l** [comment]
> It looks like Zabbix API cannot handle request. Try to tweak PHP settings and increase amount of memory assigned to PHP. I appreciate if you can make the same request in grafana with 3.x plugin version to make sure it's something that works in a different way in the new plugin versions with backend.

I tweaked up PHP settings, but grafana doesn't get metrics, only freeze and stop working during 2-5 minutes. No errors, but no metrics.

**Comment by jmrodriguezqtn** [comment]
I have the same problem, still no solution?

**Comment by alexanderzobnin** [comment]
Hm, that's strange, I will try to investigate. Can you provide a bit more info, is it happening only on large amount of data?

**Comment by hotoroy** [comment]
Hi! I have same error. I'm trying to get data on 1 item for 150 hosts.
<img width="776" alt="image" src="https://user-images.githubusercontent.com/57789614/168070352-b96aa484-fb0e-47d5-8d5b-79eb7a356178.png">

**Comment by N1k-3l** [comment]
> Hm, that's strange, I will try to investigate. Can you provide a bit more info, is it happening only on large amount of data?

Yes. As I wrote in the description, this problem only occurs when the group contains over ~50 hosts.

**Comment by GOID1989** [comment]
Have same problem on zabbix6.0.3 + grafana 8.5.2 + plugin 4.2.8

**Comment by ogovea** [comment]
I had the same problem and as a workaround, I had to create a new group in Zabbix, and apply it to hosts. It's worked for me.

**Comment by huornlmj** [comment]
I have the same issue when trying to poll a large amount of hosts with a single item per host.  If I select a time window of 3 hours I get my data, if I extend that to 6 hours I get error 500 and "no data" (error:"request failed, status: 500 Internal Server Error").  I don't know if it's Grafana or Zabbix that's throwing this 500 error as it's not showing up in either of their log files.

**Comment by huornlmj** [comment]
UPDATE - I found that if I decreased the frequency at which Zabbix polls the monitored hosts from every 1 minute to every 5 minutes and leave it for 24 hours, I can now go back to Grafana and graph those hosts up as far as 12 hours.  But anything more than 12 hours (e.g. 24 hours) and I start getting the error 500 again.  It must be to do with the volume of data being retrieved.

**Comment by rfrozza** [comment]
I was receiving the same error, but increasing the parameter `php_value memory_limit` to 1024M inside zabbix.conf file resolved the issue for me.
Tested with a dashboard with +200 hosts and 6 hours as time range 
Zabbix v6.2.3 and Grafana v9.2.5
And the setup is running on AWS, with RDS for MySQL and two EC2 instances (one for zabbix server and one for zabbix frontend).

**Comment by morelattes** [comment]
@alexanderzobnin Still facing this issue on grafana-zabbix 4.2.10, grafana 8.0 and zabbix 6.2 (tried everything mentioned in the comments). Any commits related to this?

**Comment by rmsairlock** [comment]
@alexanderzobnin same issue, grafana-zabbix 4.4.1, grafana 10.1, zabbix 6.4.6

**Comment by rmsairlock** [comment]
Just wondering, was this ever figured out? I still have the issue when getting a value from around 400 servers, if the host isn't a wild card the query/results run fine.

**Comment by Rudorufu-san** [comment]
same issue, grafana-zabbix 4.4.9, grafana 11.0, zabbix 6.4.15
